From c596b4afd544737831c4a5824b5704361dbe2cbf Mon Sep 17 00:00:00 2001
From: Iago Toral Quiroga <itoral@igalia.com>
Date: Fri, 24 Jan 2014 14:30:55 +0100
Subject: [PATCH 09/10] [WAYLAND] If bulding for both X11 and Wayland, make
 sure we use GLX to create the offscreen context when running on X11.

---
 Source/WebCore/platform/graphics/GLContext.cpp | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/Source/WebCore/platform/graphics/GLContext.cpp b/Source/WebCore/platform/graphics/GLContext.cpp
index b3ce1c0..087f63d 100644
--- a/Source/WebCore/platform/graphics/GLContext.cpp
+++ b/Source/WebCore/platform/graphics/GLContext.cpp
@@ -35,10 +35,18 @@
 #include <X11/Xlib.h>
 #endif
 
-#if USE(EGL) && PLATFORM(WAYLAND) && PLATFORM(GTK) && !defined(GTK_API_VERSION_2)
+#if PLATFORM(GTK)
+#include <gdk/gdk.h>
+#ifdef GDK_WINDOWING_WAYLAND
 #include <gdk/gdkwayland.h>
+#endif
+#ifdef GDK_WINDOWING_X11
+#include <gdk/gdkx.h>
+#endif
+#if USE(EGL) && PLATFORM(WAYLAND) && !defined(GTK_API_VERSION_2)
 #include "WaylandDisplay.h"
 #endif
+#endif
 
 using WTF::ThreadSpecific;
 
@@ -171,6 +179,12 @@ GLContext::GLContext()
 
 PassOwnPtr<GLContext> GLContext::createOffscreenContext(GLContext* sharingContext)
 {
+#if PLATFORM(GTK) && PLATFORM(WAYLAND) && PLATFORM(X11) && USE(GLX) && !defined(GTK_API_VERSION_2)
+    // We need to create context via GLX when running on X11 platform and have built for both Wayland and X11
+    GdkDisplay* display = gdk_display_manager_get_default_display(gdk_display_manager_get());
+    if (GDK_IS_X11_DISPLAY(display))
+        return GLContextGLX::createContext(0, sharingContext);
+#endif
     return createContextForWindow(0, sharingContext);
 }
 
-- 
1.8.3.2

